<?xml version="1.0"?>
<launch>

    <arg name="init_x" default="0.0"/>
    <arg name="init_y" default="0.0"/>
    <arg name="init_yaw" default="0.0"/>
    <arg name="map_file" default="$(find ndt_localizer)/example_data/d_kan_indoor.pcd"/>
    <arg name="map_frame" default="map"/>
    <arg name="use_orientation_z_as_yaw" default="false"/>

    <group ns="ndt_localizer">
        <node pkg="ndt_localizer" type="map_match" name="map_match">
            <param name="PARENT_FRAME" value="$(arg map_frame)" />
            <param name="MAP_FILE" type="string" value="$(arg map_file)"/>
            <param name="VOXEL_SIZE" value="0.3" />
            <param name="LIMIT_RANGE" value="20.0" />
            <param name="USE_ORIENTATION_Z_AS_YAW" value="$(arg use_orientation_z_as_yaw)" />
        </node>

        <node pkg="ndt_localizer" type="ekf" name="ekf">
            <rosparam file="$(find ndt_localizer)/config/ekf_sigs.yaml" />
            <param name="INIT_X" value="$(arg init_x)"/>
            <param name="INIT_Y" value="$(arg init_y)"/>
            <param name="INIT_YAW" value="$(arg init_yaw)"/>
            <param name="mode_pointing_ini_pose_on_rviz" type="bool" value="false" />
            <param name="parent_frame_name" type="string" value="$(arg map_frame)"/>
            <remap from="/imu/data" to="/imu/data" />
            <remap from="/odom" to="/odom" />
        </node>
    </group>

</launch>


